src/components/views/elements/MemberEventListSummary.js

```diff
@@ -175,61 +271,73 @@ module.exports = React.createClass({
             );
         }
 
-        // Map user IDs to the first and last member events in eventsToRender for each user
+        // Map user IDs to all of the user's member events in eventsToRender
         let userEvents = {
-            // $userId : {first : e0, last : e1}
+            // $userId : []
         };
 
-        eventsToRender.forEach((e) => {
+        eventsToRender.forEach((e, index) => {
             const userId = e.getStateKey();
             // Initialise a user's events
             if (!userEvents[userId]) {
-                userEvents[userId] = {first: null, last: null};
+                userEvents[userId] = [];
```

Comment:
Idea: push to `avatarMembers` here, instead of at line 327? I think that would have the same effect?

Then you can move the `aggregate` functionality out to another function without having to worry about populating `avatarMembers`.